### PR TITLE
Capi 959 Merge delete service workflow from internal to OSS

### PR DESCRIPTION
**Pull request by DaemonWill**
### Requirements

Problem Statement: Delete service Jenkins buildpack has gone through modularization and refactoring in internal track. We would like to bring this code into OSS.
Bring in the loadServiceMetadataModule from internal. 

Acceptance Criteria:

1. Delete service continues to work post-integration

### Description of the Change

Incorporated a new directory "jazz-build-module" which includes serviceMetadata logic. Made small modularization changes to the jenkins pipeline for delete-serverless-service-build-pack for undeployWebsite in the undeployService stage that incorporates functionality from the new module. 

### Benefits

Cleaner/better structured code

### Possible Drawbacks

modules and references would have to be accessible to the pipeline or else the build fails

### Applicable Issues

There is an issue submitted by the time of this PR in which if service/domain name is mixed case or contains uppercase characters, is-service-available will declare the service doesn't exist.

### CODE REVIEW

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 1]
```diff
@@ -773,42 +736,25 @@ def unDeployWebsite(service, domain, stage) {
  * @param stage
  * @return  true/false
  */
-def checkIfWebsiteExists(stage, folderName) {
+def checkIfWebsiteExists(bucketUrl) {
 	def status = true;
 	try {
-		sh "aws s3 ls s3://"+getBucket(stage)+"/"+folderName
+		sh "aws s3 ls s3://" + bucketUrl
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: Can you use the format in line 24 to build this string?

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -819,22 +765,18 @@ def cleanupS3BucketPolicy(service, domain, stage) {
 			sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
 			sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
 			sh "aws configure set preview.cloudfront true"
-
 			def folderName = domain + "-" + service // temporary fix
 			//def folderName = domain+"_"+service // @TODO: enable after bug fix CAPI 344
-
 			def bucketPolicy = sh (
-				script: "aws s3api get-bucket-policy --bucket " + getBucket(stage) + " --output json",
+				script: "aws s3api get-bucket-policy --bucket " + bucket + " --output json",
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: Can you use the format in line 24 to build this string?

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 3]
```diff
@@ -819,22 +765,18 @@ def cleanupS3BucketPolicy(service, domain, stage) {
...
 			).trim()
-
 			def policyObject = parseJson(parseJson(bucketPolicy).Policy)
 			def policyObjectUpdated = [:]
 			policyObjectUpdated.Version = policyObject.Version
 			policyObjectUpdated.Id = policyObject.Id
-
 			def statements = []
-			def resourcePattern = getBucket(stage)+"/"+folderName+"/"
+			def resourcePattern = bucket+"/"+folderName+"/"
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: Can you use the format in line 24 to build this string?

File: jazz-build-module/service-metadata-loader.groovy [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+@Field def g_service_name
+//@Field def g_service_region
+@Field def g_service_repository
+@Field def g_service_runtime
+//@Field def g_service_status
+//@Field def g_service_tags
+@Field def g_service_type
+//@Field def g_timestamp
+
+/* Assets schema */
```
Comment (suryajak) [inline jazz-build-module/service-metadata-loader.groovy]: Remove all commented out code

**Comment by DaemonWill** [comment]
changes made as described and tested successfully after running jenkins build

**Pull request merged**